lib/logqueue-fifo.c

```diff
@@ -343,7 +344,11 @@ log_queue_fifo_push_tail(LogQueue *s, LogMessage *msg, const LogPathOptions *pat
 
   thread_id = main_loop_worker_get_thread_id();
 
-  g_assert(thread_id < 0 || log_queue_max_threads > thread_id);
+  /* if this thread has an ID than the number of input queues we have (due
+   * to a config change), handle the load via the slow path */
+
+  if (thread_id >= self->num_input_queues)
+    thread_id = 0;
```

Comment:
[change-request] I believe this should be `-1` in order to trigger the slow path (locked per-message push).
Pushing from any non-thread0 thread to thread0's queue would not be safe as far as I see.

Reply:
You are right. That should be -1.

There's a -1 dance scattered everywhere in the code, sometimes the "id" is already a zero-based value, sometimes is 1 based.

I am thinking about naming one an "id" (the 1 based one), and the other "index" (the zero based one), so it's easy to understand which one is which.